framework/include/fwk_interrupt.h

```diff
@@ -187,6 +196,18 @@ int fwk_interrupt_set_isr_param(unsigned int interrupt,
  */
 int fwk_interrupt_get_current(unsigned int *interrupt);
 
+/*!
+ * \brief Set the fault interrupt service routine. This is used internally by
+ *      the framework.
+ *
+ * \param isr Pointer to the fault interrupt service routine function.
+ *
+ * \retval FWK_SUCCESS Operation succeeded.
+ * \retval FWK_E_PARAM One or more parameters were invalid.
+ * \retval FWK_E_INIT The component has not been initialized.
+ */
+int fwk_interrupt_set_isr_fault(void (*isr)(void));
```

Comment:
Both these calls are used exclusively by the Framework only for internal purposes. I see currently we have externed them and used them. However, we should not put these declarations here, since modules also might use these files. Would you consider putting them in an internal header perhaps?

Reply:
Ok, I'll see how to move into internal `fwk_interrupt.h` header file.

Reply:
I think this function can be removed entirely - it isn't used anywhere except the tests, and the tests are testing... this function. It is, as far as I can see, entirely pointless, especially given the comment:

```
/* This function is only for internal use by the framework */
```

Which seems to be untrue. `fwk_arch_interrupt_driver::set_isr_fault` and the functions that use it also needs removing. It's up to the architecture layer to define proper ISR fault routines, as the framework cannot determine what is safe to do within the fault, nor which fault it even is.